pkg/api/steve/supportconfigs/generator_test.go

```diff
@@ -0,0 +1,182 @@
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests PAYG, authorized to get PAYG, PAYG not installed (501)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests PAYG, not authorized to get PAYG, auth denied (403)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, BYOL not installed (501)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests BYOL, not authorized to get BYOL, auth denied (403)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, we return output (200)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+		{
+			name:                    "test user requests PAYG, autorized to get PAYG, we return output (200)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+	}
+	for _, scenario := range scenarios {
+		test := scenario
+		t.Run(test.name, func(t *testing.T) {
+			h := &Handler{
+				adapterUtil: NewFakeChartUtil(test.generateAdapterError, test.generateAdapterNotFound),
+				cspConfig:   NewFakeCSPConfig(test.generateAuthorizedError, test.authorized, test.marshalledCSPConfig),
+			}
+			reqPath := "/v1/generateSUSERancherSupportConfig"
+			if test.usePAYG {
+				reqPath = reqPath + "?usePAYG=true"
+			}
+			req := httptest.NewRequest(http.MethodGet, reqPath, nil)
+			rr := httptest.NewRecorder()
+			h.ServeHTTP(rr, req)
+			assert.Equal(t, test.expectedHTTPCode, rr.Code)
```

Comment:
There isn't a test or assertion that inspects the generated supportconfig in a successful response. @MbolotSuse, let me know if this is outside the scope of this PR, but might be important enough for the future.

Also, @MbolotSuse, let me know if we want to test the actual methods of `CSPConfig` with its fake dependencies, rather than a completely different fake type that implements its methods in its own way.

Reply:
> @MbolotSuse, let me know if this is outside the scope of this PR, but might be important enough for the future. 

I would consider this out of scope for this PR - @guangyee did not change that logic (excepting the namespace lookup), so I think it's ok for him to not have to write the tests for that.

> let me know if we want to test the actual methods of CSPConfig with its fake dependencies, rather than a completely different fake type that implements its methods in its own way.

I agree that we want to test the actual methods, will leave a comment detailing how to achieve this.

Reply:
We are not actually testing the content of the configmap itself right as the changes doesn't impact that part of the code?